kube-core/src/schema.rs

```diff
@@ -63,6 +64,15 @@ impl Visitor for StructuralSchemaRewriter {
                     .insert("x-kubernetes-preserve-unknown-fields".into(), true.into());
             }
         }
+
+        // As of version 1.30 Kubernetes does not support setting `uniqueItems` to `true`,
+        // so we need to remove this fields.
+        // TODO: Ideally we would set `x-kubernetes-list-type` instead, but the question is
+        // to what value (set or map) we should set it, see https://kubernetes.io/docs/reference/using-api/server-side-apply/
+        // for details.
+        if let Some(array) = &mut schema.array {
+            array.unique_items = None;
+        }
```

Comment:
I guess this is ultimately a property for validation on the apiserver side so this would more be for getting the default case to work. If people want that type of validation they can still inject `x-kubernetes-list-type: set`, but the docs say this only works "for scalars" so probably not something we can set here. This seems like a sensible start. Remove the unsupported.

Reply:
I changed the TODO section in 55a740f19bdc5c51bce5282aba6dbeaa754a087b, I hope that fits better